modules/signals/spec/patch-state.spec.ts

```diff
@@ -74,4 +76,20 @@ describe('patchState', () => {
     expect(state.numbers()).not.toBe(initialState.numbers);
     expect(state.ngrx()).not.toBe(initialState.ngrx);
   });
+
+  it(
+    'should not emit if there was no change',
+    testEffects((tick) => {
+      let emitCount = 0;
+      const state = signalState(initialState);
+      effect(() => emitCount++);
+
+      tick();
+      expect(emitCount).toBe(1);
+      patchState(state, {});
+      patchState(state, (state) => state);
+
+      expect(emitCount).toBe(1);
```

Comment:
if we read the `state` value in the `effect` and invoke `tick` after `patchState`:

```suggestion
      effect(() => {
        state();
        emitCount++;
      });

      tick();
      expect(emitCount).toBe(1);
      patchState(state, {});
      patchState(state, (state) => state);
      tick();
      
      expect(emitCount).toBe(1);
```

this test will fail because the state value reference will be changed. However, nested signals will not emit. Here is the test that verifies this behavior: https://github.com/ngrx/platform/blob/main/modules/signals/spec/signal-state.spec.ts#L54

I suggest moving this test to the `signal-state.spec.ts` and verifying that nested signals don't emit when the signal state is patched with an empty object/updater that doesn't change the state.

Reply:
Done, please see my remarks below.